### PR TITLE
[HLS-425] Remove dependency to GATK's SimpleInterval

### PR DESCRIPTION
**Pull request by kianfar77**
## What changes are proposed in this pull request?
This PR replaces the dependency to ``org.broadinstitute.hellbender.utils.SimpleInterval`` in Glow with an internal SimpleInterval class. 

## How is this patch tested?
- [x] Unit tests
- [x] Integration tests
- [x] Manual tests

(Details)

### CODE REVIEW

File: core/src/main/scala/io/projectglow/common/GenomicIntervalUtils.scala [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+package io.projectglow.common
+
+import scala.math.{max, min}
+
+object GenomicIntervalUtils {
+
+  /**
+   * A class representing a closed interval of longs that also accommodates an empty interval.
+   */
+  class LongInterval() extends Serializable {
```
Comment (henrydavidge) [inline core/src/main/scala/io/projectglow/common/GenomicIntervalUtils.scala:26]: It seems like this should just be a
```scala
case class LongInterval(coordinates: Option[(Long, Long)])
```

Then you get `isSame` (which should just be `equals`), `isEmpty`, and `nonEmpty` for free.

File: core/src/main/scala/io/projectglow/common/GenomicIntervalUtils.scala [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    var interval: Option[(Long, Long)] = None
+
+    def this(start: Long, end: Long) {
+      this()
+      if (end >= start) {
+        this.interval = Option((start, end))
+      }
+    }
+
+    def start: Long =
```
Comment (henrydavidge) [inline core/src/main/scala/io/projectglow/common/GenomicIntervalUtils.scala:36]: IMO it's bad practice for simple accessors like this to throw exceptions. It's better to just expose method that gets the optional start. I think that the simple `coordinates` covers that.

File: core/src/main/scala/io/projectglow/vcf/TabixIndexHelper.scala [hunk 3]
```diff
@@ -27,151 +27,52 @@ import org.apache.hadoop.conf.Configuration
...
-        i.getContig == j.getContig && i.getStart == j.getStart && i.getEnd == j.getEnd
-      case (None, _) => other.isEmpty
-      case (_, None) => interval.isEmpty
+case class ParsedFilterResult(
+    contig: Contig,
+    startInterval: LongInterval,
+    endInterval: LongInterval)
+
+case class ContigAndInterval(contig: Contig, interval: LongInterval) extends Serializable {
+  def toHTSJDKInterval: Interval = {
```
Comment (henrydavidge) [inline core/src/main/scala/io/projectglow/vcf/TabixIndexHelper.scala:41]: This should return an option.

**Comment by kianfar77** [comment]
Based on your comments, I think you prefer minimal change to existing code. Revised for minimal change.

**Pull request merged**